app/feature/feature-changeaddress/src/main/kotlin/com/hedvig/android/feature/changeaddress/destination/offer/ChangeAddressOfferViewModel.kt

```diff
@@ -0,0 +1,152 @@
+package com.hedvig.android.feature.changeaddress.destination.offer
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.hedvig.android.feature.changeaddress.data.ChangeAddressRepository
+import com.hedvig.android.feature.changeaddress.data.MoveIntentId
+import com.hedvig.android.feature.changeaddress.data.MoveQuote
+import com.hedvig.android.feature.changeaddress.data.SuccessfulMove
+import com.hedvig.android.feature.changeaddress.destination.createQuoteInput
+import com.hedvig.android.feature.changeaddress.destination.offer.ChangeAddressOfferEvent.ConfirmMove
+import com.hedvig.android.feature.changeaddress.destination.offer.ChangeAddressOfferEvent.DismissErrorDialog
+import com.hedvig.android.feature.changeaddress.destination.offer.ChangeAddressOfferEvent.ExpandQuote
+import com.hedvig.android.feature.changeaddress.destination.offer.ChangeAddressOfferEvent.Retry
+import com.hedvig.android.feature.changeaddress.navigation.MovingParameters
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.datetime.LocalDate
+
+internal class ChangeAddressOfferViewModel(
+  previousParameters: MovingParameters,
+  changeAddressRepository: ChangeAddressRepository,
+) : MoleculeViewModel<ChangeAddressOfferEvent, ChangeAddressOfferUiState>(
+    initialState = ChangeAddressOfferUiState(
+      movingDate = previousParameters.newAddressParameters.movingDate,
+      moveIntentId = MoveIntentId(previousParameters.selectHousingTypeParameters.moveIntentId),
+    ),
+    presenter = ChangeAddressOfferPresenter(
+      previousParameters = previousParameters,
+      changeAddressRepository = changeAddressRepository,
+    ),
+  )
+
+internal class ChangeAddressOfferPresenter(
+  private val previousParameters: MovingParameters,
+  private val changeAddressRepository: ChangeAddressRepository,
+) : MoleculePresenter<ChangeAddressOfferEvent, ChangeAddressOfferUiState> {
+  @Composable
+  override fun MoleculePresenterScope<ChangeAddressOfferEvent>.present(
+    lastState: ChangeAddressOfferUiState,
+  ): ChangeAddressOfferUiState {
+    var currentState by remember { mutableStateOf(lastState) }
+
+    var dataLoadIteration by remember { mutableIntStateOf(0) }
+
+    var quotesLoadIteration by remember { mutableIntStateOf(0) }
+
+    CollectEvents { event ->
+      when (event) {
+        is ConfirmMove -> {
+          dataLoadIteration++
```

Comment:
Since this is a bit more than just refreshing the most up-to-date data to show in the UI. Could we name it something a bit more descriptive. Like the confirmMoveIteration or something like that?